handler.js

```diff
@@ -19,74 +24,87 @@ var region = process.env.AWS_DEFAULT_REGION;
 // this function will be triggered by the github webhook
 module.exports.start_build = (event, context, callback) => {
 
-  var response;
+  var response = {
+    pull_request: {},
+    build: {}
+  };
 
   // we only act on pull_request changes (can be any, but we don't need those)
   if('pull_request' in event) {
 
-    var head = event.pull_request.head;
-    var repo = head.repo;
-
-    var params = {
-      projectName: process.env.BUILD_PROJECT,
-      sourceVersion: head.sha
-    };
-
-    // start the codebuild process for this project
-    codebuild.startBuild(params, function(err, data) {
-      if (err) {
-        console.log(err, err.stack);
-        callback(err);
-      } else {
-
-        var build = data.build;
-
-        // all is well, mark the commit as being 'in progress'
-        github.repos.createStatus({
-          owner: repo.owner.login,
-          repo: repo.name,
-          sha: head.sha,
-          state: 'pending',
-          target_url: 'https://' + region + '.console.aws.amazon.com/codebuild/home?region=' + region + '#/builds/' + data.build.id + '/view/new',
-          context: 'CodeBuild',
-          description: 'Build is running...'
-        }).then(function(data){
-          console.log(data);
-        });
-        callback(null, build);
-      }
-    });
+    if(BUILD_ACTIONS.indexOf(event.action) >= 0) {
+
+      response.pull_request = event.pull_request
+      var head = event.pull_request.head;
+      var base = event.pull_request.base;
+      var repo = base.repo;
+
+      var params = {
+        projectName: process.env.BUILD_PROJECT,
```

Comment:
Ok, that only works if the name of the codebuild project is the same as the repository name, which is not necessarily the case. Since we use this env in serverless.yml anyway, I don't see the harm in passing it through to the lambda function.